### DISCUSSION TITLE
Deprovision users in application

### DISCUSSION THREAD

**Discussion by sr258** [goauthentik/authentik#4298] [Q&A] [answered]
I'm currently evaluating authentik for my project. I want to manage users in authentik, but obviously my main application stores user data. If an account is deleted in authentic, the user data in the application must be deleted as well. Does authentik support this? I think a simple way would be to call a webhook whenever a user is deleted. The documentation on event transports with webhook isn't very clear on whether the data that is sent to the hook can be used for my use case.

**Comment by BeryJu** [comment, marked as answer]
Yeah you can use the event system to do this, on the `model_deleted` event for the model `user` in the app `authentik_core`, send a request. This blog post has a bit bigger of an example on how to send webhooks from event policies: https://beryju.io/blog/2022-05-authentik-vault/

**Reply by sr258** [reply]
I now have a running authentik instance and could test it. Works fine! Thanks!

**Reply by local-it-dev** [reply]
I try to implement the same, but unfortunately the `model_deleted` event does not contain enough information about the model to send API requests for the user deletion. The `model_deleted` event only contains the `Name` but neither the username nor the mail address, which is used by many apps to identify the user. It also contains the `pk` but as the model is already deleted I can not use it to get the other user information.
